docs/tutorials/parsec-tutorial.md

```diff
@@ -0,0 +1,582 @@
+    path = 'disk-image/parsec/parsec-image/parsec',
+    inputs = [packer, experiments_repo, m5_binary, parsec_repo,],
+    documentation = 'Ubuntu with m5 binary and PARSEC installed.'
+)
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'gem5 binary'
+)
+
+linux_repo = Artifact.registerArtifact(
+    command = '''git clone --branch v4.19.83 --depth 1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git;
+    mv linux linux-stable''',
+    typ = 'git repo',
+    name = 'linux-stable',
+    path =  'linux-stable/',
+    cwd = './',
+    documentation = 'linux kernel source code repo'
+)
+
+linux_binary = Artifact.registerArtifact(
+    name = 'vmlinux-4.19.83',
+    typ = 'kernel',
+    path = 'linux-stable/vmlinux-4.19.83',
+    cwd = 'linux-stable/',
+    command = '''
+    cp ../config.4.19.83 .config;
+    make -j8;
+    cp vmlinux vmlinux-4.19.83;
+    ''',
+    inputs = [experiments_repo, linux_repo,],
+    documentation = "kernel binary for v4.19.83"
+)
+```
+
+Once, all of the artifacts are registered, the next step is to launch all gem5 jobs. To do that, add the following lines in your script:
+
+```python
+if __name__ == "__main__":
+    num_cpus = ['1']
+    benchmarks = ['blackscholes', 'bodytrack', 'canneal', 'dedup','facesim', 'ferret', 'fluidanimate', 'freqmine', 'raytrace', 'streamcluster', 'swaptions', 'vips', 'x264']
+
+    sizes = ['simsmall', 'simlarge', 'native']
+    cpus = ['kvm', 'timing']
+
+    for cpu in cpus:
+        for num_cpu in num_cpus:
+            for size in sizes:
+                if cpu == 'timing' and size != 'simsmall':
+                    continue
+                for bm in benchmarks:
+                    run = gem5Run.createFSRun(
+                        'parsec_tests',    
+                        'gem5/build/X86/gem5.opt',
+                        'configs-parsec-tests/run_parsec.py',
+                        f'''results/run_parsec/{bm}/{size}/{cpu}/{num_cpu}''',
+                        gem5_binary, gem5_repo, experiments_repo,
+                        'linux-stable/vmlinux-4.19.83',
+                        'disk-image/parsec/parsec-image/parsec',
+                        linux_binary, disk_image,
+                        cpu, bm, size, num_cpu,
+                        timeout = 24*60*60 #24 hours
+                        )
+                    run_gem5_instance.apply_async((run, os.getcwd(), ))
+```
+The above lines are responsible for looping through all possible combinations of variables involved in this experiment.
+For each combination, a gem5Run object is created and eventually passed to run_gem5_instance to be
+executed asynchronously using Celery.
+
+
+Finally, make sure you are in python virtual env and then run the script:
+
+```python
+python launch_parsec_tests.py
+```
+
+## Results
```

Comment:
Do we have the results? We don't need to put off merging this for the results, but I'd like to have them for gem5-19, if possible